docs/source/contributing.rst

```diff
@@ -0,0 +1,187 @@
+
+- ``KordacResult()``
+    The object returned by ``Kordac()`` containing:
+    	- Converted html string
+    	- Title
+    	- Required files (images, interactives, scratch images, page scripts)
+    	- Heading tree
+    	- Required glossary terms
+
+- ``KordacExtension()``
+    Inherits the ``Extension`` class from Markdown.
+    This class is the main class of the project. It loads all the processor information, loads the template files and clears and populates the attributes to be returned by the ``KordacResult`` object.
+
+- ``Processors/``
+  	There is a different processor for each tag. A processor uses it's corresponding regex loaded from ``processor-info.json`` to find matches in the text, and uses the given arguments in the matched tag to populate and output it's html template.
+
+- ``html-templates/``
+  	The html templates (using Jinja2 template engine) to be populated by processors.
+
+- ``processor-info.json``
+	Every processor is listed in this file, and will at least contain a regex pattern to match it's corresponding tag.
+	Most will also define required and optional parameters, these correspond to arguments in the tag's template.
+
+
+
+Creating a New Processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To create a new processor, a good place to start is the `Extension API`_ page of the Python Markdown docs, or even the `source code`_ itself.
+
+.. _Extension API: https://pythonhosted.org/Markdown/extensions/api.html
+
+.. _source code: https://github.com/waylan/Python-Markdown
+
+
+There are several different kinds of processors in Python Markdown, each serving a slightly different purpose. We recommend reading the API docs to determine which processor best suits your purpose (Kordac currently makes use of ``preprocessor``, ``blockprocessor``, ``inlinepattern`` and ``treeprocessor``).
+
+The order of the processors matters and is defined when each processor is added to the ``OrderedDict`` in ``KordacExtension.py``.
+
+Each processor is independent of every other processor. If you have two processors in the pipeline that may overlap (e.g. codehilite and fencedcode), the second processor must handle whatever the first outputs, i.e. refrain from manipulating the output of the first processor to help the second.
+
+Every processor belongs in the ``processors/`` directory, but there are several other places they need to be listed, these are:
+
+- The frozenset of ``DEFAULT_PROCESSORS`` in ``Kordac.py``
+- The relevant list in ``extendMarkdown()`` in ``KordacExtension.py`` (see `OrderedDict in the Markdown API docs`_ for determining order)
+- The processor's template should be added to ``html-templates`` using the Jinja2 Template Engine syntax for variable parameters
+- The processor's relevant information (regex pattern, required parameters etc) should be included in ``processor-info.json``
+
```

Comment:
Also could note about adding to default list if it should be enabled by default.